elliottlib/errata_async.py

```diff
@@ -91,51 +92,114 @@ async def delete_cve_package_exclusion(self, exclusion_id: int):
+        # for now we skip etcd and associate every cve with all base_golang build images
+        # TODO: figure out how to find out if a golang CVE only affects etcd
+
+        if etcd_golang_builder:
+            _LOGGER.warning(f'etcd build found in advisory {go_nvr_map[etcd_golang_builder][0]}, with builder: '
+                            f'{etcd_golang_builder}. If an attached CVE affects etcd please manually associate CVE '
+                            'with etcd build')
+
+        for cve_name in golang_cve_names:
+            expected_cve_components[cve_name] = {nvr[0] for nvr in go_nvr_map[base_golang_builder]}
+            _LOGGER.info(f"Associating golang {cve_name} with all golang "
+                         f"images ({len(go_nvr_map[base_golang_builder])})")
+
+        return expected_cve_components
+
+    @classmethod
+    def diff_cve_exclusions(cls, current_exclusions: Dict[str, Dict[str, int]],
+                            expected_exclusions: Dict[str, Dict[str, int]]):
         """ Given 2 cve_package_exclusions dicts, return the difference.
-        :return: (extra_cve_package_exclusions, missing_cve_package_exclusions)
+        :return: (extra_exclusions, missing_exclusions)
         """
-        extra_cve_package_exclusions = {cve: {pkg: exclusions[pkg] for pkg in exclusions.keys() - expected_cve_packages_exclusions.get(cve, {}).keys()} for cve, exclusions in current_cve_package_exclusions.items()}
-        missing_cve_package_exclusions = {cve: {pkg: exclusions[pkg] for pkg in exclusions.keys() - current_cve_package_exclusions.get(cve, {}).keys()} for cve, exclusions in expected_cve_packages_exclusions.items()}
-        return extra_cve_package_exclusions, missing_cve_package_exclusions
+        extra_exclusions = {cve: {pkg: exclusions[pkg] for pkg in exclusions.keys()
+                                  - expected_exclusions.get(cve, {}).keys()}
+                            for cve, exclusions in current_exclusions.items()}
+        missing_exclusions = {cve: {pkg: exclusions[pkg] for pkg in exclusions.keys()
+                                    - current_exclusions.get(cve, {}).keys()}
+                              for cve, exclusions in expected_exclusions.items()}
+        return extra_exclusions, missing_exclusions
 
     @classmethod
-    async def associate_builds_with_cves(cls, api: AsyncErrataAPI, advisory_id: int, attached_builds: List[str], cve_components_mapping: Dict[str, Set[str]], dry_run=False):
-        """ Request Errata to associate CVEs to attached Brew builds
-        :param api: Errata API
-        :param advisory_id: advisory id
-        :param attached_builds: list of attached Brew build NVRs
-        :param cve_components_mapping: a dict mapping each CVE to a list of brew components
-        """
+    async def validate_cves_and_get_exclusions_diff(cls, api: AsyncErrataAPI, advisory_id: int, attached_builds: List[
```

Comment:
Extract out the validation piece so we can reuse it in verify_attached_bugs_cli.py